component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpRequestSink.java

```diff
@@ -485,15 +699,33 @@ public void publish(Object payload, DynamicOptions dynamicOptions) {
         }
         cMessage.completeMessage();
         HttpResponseFuture httpResponseFuture = clientConnector.send(cMessage);
+        CountDownLatch latch = new CountDownLatch(1);
         HttpResponseMessageListener httpListener =
-                new HttpResponseMessageListener(getTrpProperties(dynamicOptions), sinkId, isDownloadEnabled);
+                new HttpResponseMessageListener(getTrpProperties(dynamicOptions), sinkId, isDownloadEnabled, latch,
+                        tryCount, authType);
         httpResponseFuture.setHttpConnectorListener(httpListener);
+        if (HttpConstants.OAUTH.equals(authType)) {
+            try {
+                boolean latchCount = latch.await(30, TimeUnit.SECONDS);
+                if (!latchCount) {
+                    log.debug("Time out due to getting getting response from " + publisherURL);
+                    throw new HttpSinkAdaptorRuntimeException("Time out due to getting getting response from "
+                            + publisherURL);
+                }
+            } catch (InterruptedException e) {
+                log.debug("Time out due to getting getting response from " + publisherURL + e);
```

Comment:
Here, the thread has not gotten a chance to wait till the latch is counted down, as it was interrupted. In that sense, this debug log is misleading.